detection_rules/utils.py

```diff
@@ -44,18 +43,17 @@ def get_json_iter(f):
 
 def get_path(*paths):
```

Comment:
we could probably get rid of all these functions.
we basically reimplemented pathlib, only worse

Reply:
I see the tests failed after `devtools.license_check` called `glob` on a path object. I can try and refactor it throughout to use `pathlib` if you and @brokensound77 are good with that.

Reply:
i'm game for a complete pathlib refactor if you're up for the challenge.
just know you've opened pandora's box.

but i'm thinking... pretty much all of these files can be deleted. idk if pathlib has json/yaml capabilities, but we could refactor those functions to take a Path object instead

Reply:
I'll dig into it a bit more and give it another go. Cheers

Reply:
You weren't joking 😅

Reply:
🤣